ExposureNotification/ViewControllers/UploadKeysViewController.swift

```diff
@@ -415,6 +407,25 @@ extension UploadKeysViewController: UITextFieldDelegate {
     }
 }
 
+extension UploadKeysViewController: UIGestureRecognizerDelegate {
+    func gestureRecognizerShouldBegin(_ gestureRecognizer: UIGestureRecognizer) -> Bool {
+        
+        // The views that should not be responsed to the tap gesture.
+        let whiteListViews = [startDatePicker, endDatePicker]
+        
+        let location = gestureRecognizer.location(in: gestureRecognizer.view)
+        
+        for view in whiteListViews {
+            let point = view.convert(location, from: gestureRecognizer.view)
+            if view.point(inside: point, with: nil) {
+                return false
+            }
+        }
+        
+        return true
```

Comment:
I think we don't need the `location`. How about converting location directly onto `whiteListViews`? Something like

```swift
for view in whiteListViews {
    let point = gestureRecognizer.location(in: view)
    ...
}
```

And think functional, we can do like following

```swift
        return whiteListViews.map {
            let location = gestureRecognizer.location(in: $0)
            // Break the chain if touching inside views of white list.
            return $0.point(inside: location, with: nil) == false
        }.reduce(true) { $0 && $1 }
```

Reply:
Yes, don't need the `convert` function. And I have summited the new code (more functional).

Reply:
Actually, for the functions that need to return directly, I think the original code may be better, reducing the times of map the array.

Reply:
You use `allSatisfy` which is much better than I suggested. Thanks!